lib/std/c/haiku.zig

```diff
@@ -3,17 +3,47 @@
 // This file is part of [zig](https://ziglang.org/), which is MIT licensed.
 // The MIT license requires this copyright notice to be included in all copies
 // and substantial portions of the software.
+
+//
+const std = @import("../std.zig");
+const builtin = std.builtin;
+
+usingnamespace std.c;
+
+extern "c" fn _errnop() *c_int;
+
+pub const _errno = _errnop;
+
+pub extern "c" fn _kern_read_dir(fd: c_int, buf_ptr: [*]u8, nbytes: usize, maxcount: u32) usize;
```

Comment:
`_kern...` functions are syscall hooks, and we make no guarantees whatsoever about their ABIs, which are subject to change without notice, etc. (Calling syscalls via hooks is at least better than by number, though, as the numbers are generated at build time and so change whenever syscalls are touched in pretty much any way.)

Reply:
thanks! i will keep that in mind, i think some of these exposed syscall hooks might not be as necessary when _dl_iterat_phdr_ support is available